### ISSUE TITLE
Distributed tests hang with nccl 2.13.4

### ISSUE THREAD

**Issue by emcastillo** [cupy/cupy#6912]
### Description

Before this CI run in cuda 11.7
https://ci.preferred.jp/cupy.linux.cuda117.multi/106761/

NCCL was not installed, but after nccl 2.13.4 was installed in https://ci.preferred.jp/cupy.linux.cuda117.multi/106761/

distributed tests are failing.

```


00:22:12.655634 STDOUT 1614] | OS                           : Linux-5.11.0-1023-gcp-x86_64-with-glibc2.31 |
00:22:12.655636 STDOUT 1614] | Python Version               : 3.10.0 |  
00:22:12.655644 STDOUT 1614] | CuPy Version                 : 11.0.0rc1 |  
00:22:12.655670 STDOUT 1614] | CuPy Platform                : NVIDIA CUDA |  
00:22:13.009264 STDOUT 1614] | NumPy Version                : 1.23.1 |  
00:22:13.009373 STDOUT 1614] | SciPy Version                : 1.8.1 |  
00:22:13.009375 STDOUT 1614] | Cython Build Version         : 0.29.30 |  
00:22:13.009379 STDOUT 1614] | Cython Runtime Version       : 0.29.30 |  
00:22:13.009451 STDOUT 1614] | CUDA Root                    : /usr/local/cuda |  
00:22:13.009454 STDOUT 1614] | nvcc PATH                    : ccache nvcc |  
00:22:13.009459 STDOUT 1614] | CUDA Build Version           : 11070 |  
00:22:13.009517 STDOUT 1614] | CUDA Driver Version          : 11070 |  
00:22:13.009520 STDOUT 1614] | CUDA Runtime Version         : 11070 |  
00:22:13.009524 STDOUT 1614] | cuBLAS Version               : 111001 |  
00:22:13.009550 STDOUT 1614] | cuFFT Version                : 10702 |  
00:22:13.009552 STDOUT 1614] | cuRAND Version               : 10210 |  
00:22:13.009556 STDOUT 1614] | cuSOLVER Version             : (11, 3, 5) |  
00:22:13.009600 STDOUT 1614] | cuSPARSE Version             : 11703 |  
00:22:13.009602 STDOUT 1614] | NVRTC Version                : (11, 7) |  
00:22:13.009606 STDOUT 1614] | Thrust Version               : 101500 |  
00:22:13.009667 STDOUT 1614] | CUB Build Version            : 101500 |  
00:22:13.009669 STDOUT 1614] | Jitify Build Version         : <unknown> |  
00:22:13.009673 STDOUT 1614] | cuDNN Build Version          : 8401 |  
00:22:13.009703 STDOUT 1614] | cuDNN Version                : 8401 |  
00:22:13.009704 STDOUT 1614] | NCCL Build Version           : 21304 |  
00:22:13.009708 STDOUT 1614] | NCCL Runtime Version         : 21304 |  
00:22:13.009740 STDOUT 1614] | cuTENSOR Version             : 10500 |  
00:22:13.009742 STDOUT 1614] | cuSPARSELt Build Version     : 200 |  
00:22:13.009745 STDOUT 1614] | Device 0 Name                : Tesla T4 |  
00:22:13.009772 STDOUT 1614] | Device 0 Compute Capability  : 75 |  
00:22:13.009774 STDOUT 1614] | Device 0 PCI Bus ID          : 0000:00:04.0 |  
00:22:13.009777 STDOUT 1614] | Device 1 Name                : Tesla T4 |  
00:22:13.009812 STDOUT 1614] | Device 1 Compute Capability  : 75 |  
00:22:13.009814 STDOUT 1614] | Device 1 PCI Bus ID          : 0000:00:05.0
```

**Comment by kmaehashi** [comment]
Seems this is the same one we observed in #6534

**Comment by kmaehashi** [comment]
NCCL seems to be added to the base nvidia/cuda image 6 days ago

**Comment by emcastillo** [comment]
Fixed by #6942
